### PR TITLE
raftstore-v2: update region state in apply_snapshot

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Ref #12842 

What's Changed:

```commit-message
update region state in apply_snapshot
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore-v2/src/operation/ready/snapshot.rs [hunk 1]
```diff
@@ -596,12 +592,15 @@ impl<EK: KvEngine, ER: RaftEngine> Storage<EK, ER> {
...
         );
-        let region_state = self.region_state_mut();
+        let mut region_state = self.region_state().clone();
         region_state.set_state(PeerState::Normal);
         region_state.set_region(region);
         region_state.set_removed_records(removed_records);
         region_state.set_merged_records(merged_records);
         region_state.set_tablet_index(last_index);
+        // We need set_region_state here to update the peer.
+        self.set_region_state(region_state);
```
Comment (BusyJay) [inline components/raftstore-v2/src/operation/ready/snapshot.rs:602]: I don't get it. So if L599 already change the state in place, why need to set it again in L263?

Comment (SpadeA-Tang) [reply components/raftstore-v2/src/operation/ready/snapshot.rs:602]: `set_region_state` does not just set the region_state, it also update the peer.

**Review by BusyJay — APPROVED** [review]

**Review by tabokie — APPROVED** [review]

**Pull request merged**